### PR TITLE
Implement a REQUEUE command

### PR DESCRIPTION
**Pull request by lackstein**
Implements a `REQUEUE {"jid": "..."}` command that ACKs the given job and then immediately requeues it to the beginning of the queue. This can be useful if you need a worker to give up execution of a job without impacting the retry count, for example if the worker is scaling down.

### CODE REVIEW

File: cmd/faktory/daemon.go
Comment (jagonalez) [inline cmd/faktory/daemon.go:16]: wow github actions shows the actual error now, that's super cool

File: requeue/requeue.go
Comment (jagonalez) [inline requeue/requeue.go:88]: i was comparing this to what faktory does, and it looks like a more recent version added a change: https://github.com/contribsys/faktory/commit/1e4d823ee12da8e46dc7f221984bc86767482cc0

Can we see if including still functions properly? otherwise we'll need to make a note to update the code when we upgrade the faktory dependency.

Comment (lackstein) [reply requeue/requeue.go:88]: Ughhhh good catch

**Review by jagonalez — APPROVED** [review]
code LGTM - let me know if you want me to test it, but the unit tests should suffice

File: requeue/requeue.go
Comment (lackstein) [reply requeue/requeue.go:88]: This change looks like it's going to require us to rewrite all of our middleware since the type of `MiddlewareFunc` changed. I guess we'll cross that bridge when the next version of Faktory is released.

**Review by elldritch — APPROVED** [review]
I like your tests!

LGTM (passes tests as is), but small improvements I'd like you to make before you merge.

**Pull request merged**
